PhysicsTools/NanoAOD/plugins/GenWeightsTableProducer.cc

```diff
@@ -763,21 +842,93 @@ class GenWeightsTableProducer : public edm::global::EDProducer<edm::StreamCache<
   }
 
   // create an empty counter
-  std::unique_ptr<CounterMap> beginStream(edm::StreamID) const override { return std::make_unique<CounterMap>(); }
+  std::unique_ptr<LumiCacheInfoHolder> beginStream(edm::StreamID) const override { return std::make_unique<LumiCacheInfoHolder>(); }
   // inizialize to zero at begin run
   void streamBeginRun(edm::StreamID id, edm::Run const&, edm::EventSetup const&) const override {
     streamCache(id)->clear();
   }
   void streamBeginLuminosityBlock(edm::StreamID id,
                                   edm::LuminosityBlock const& lumiBlock,
                                   edm::EventSetup const& eventSetup) const override {
-    auto counterMap = streamCache(id);
+    auto counterMap = &(streamCache(id)->countermap);
     edm::Handle<GenLumiInfoHeader> genLumiInfoHead;
     lumiBlock.getByToken(genLumiInfoHeadTag_, genLumiInfoHead);
     if (!genLumiInfoHead.isValid())
       edm::LogWarning("LHETablesProducer")
           << "No GenLumiInfoHeader product found, will not fill generator model string.\n";
-    counterMap->setLabel(genLumiInfoHead.isValid() ? genLumiInfoHead->configDescription() : "");
+    std::string label;
+    if (genLumiInfoHead.isValid()) {
+      label = genLumiInfoHead->configDescription();
+      boost::replace_all(label,"-","_");
+    }
+    counterMap->setLabel(label);
+    
+    if (!genLumiInfoHead.isValid()) return;
```

Comment:
please put all the following inside a proper `if` clause, instead of introducing a `return` that can makes future development of this code more tricky

Reply:
done